src/_utils/useReducedMotion/index.tsx

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from 'react'
+
+/**
+ * This custom hook will return reduced motion boolean
+ * To be used to disable animation based on user preferences
+ * https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion
+ *
+ */
+export const useReducedMotion = () => {
```

Comment:
I'm still confused by these global helper hooks with useEffect.
I would prefer something like a global util:
const USE_REDUCED_MOTION = window.matchMedia('(prefers-reduced-motion)').matches
(we can even cache the result if really needed)

This is simpler to read and does not depend on React lifecycle/hooks (it does not use useEffect).

To me it's not very valuable to have this as a global hook and even misleading: It should be called useReducedMotionPostRender to match what it's actually doing.
Am I missing some use cases ?

Reply:
> const USE_REDUCED_MOTION = window.matchMedia('(prefers-reduced-motion)').matches

Reply:
Ok let's rename the hook to have a correct name using some suffix (postRender, onMount, etc)

Reply:
Maybe:

```ts
const hasReducedMotion = () => {
  if (window == null) {
    return false
  }

  return window.matchMedia('(prefers-reduced-motion)').matches
}

export const useReducedMotion = () => {
  // During server side the value is `false`, during client side rendering it is initialised with
  // the correct value.
  const [reducedMotion, setReducedMotion] = useState(hasReducedMotion)

  useEffect(() => {
    // If the component was serve-sider rendered we update the value.
    // Otherwise the state won't change
    setReducedMotion(hasReducedMotion())
  }, [])

  return reducedMotion
}
```

Reply:
> To me it's not very valuable to have this as a global hook and even misleading: It should be called useReducedMotionPostRender to match what it's actually doing.

I don't see the point of renaming or even changing the code. As a hook consumer you want a value from it and use it in your component code, for me what the hook is doing and how is not important IMHO. 

@grosbouddha why do you need to know it will be run post render?